### PR TITLE
Mapper creation

### PR DESCRIPTION
**Pull request by dominikkukacka**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/EnoF/con-rest/pull/45%23issuecomment-69584159%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23issuecomment-69704812%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23issuecomment-69717378%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23issuecomment-69717464%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23discussion_r22846128%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23discussion_r22846246%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23discussion_r22851067%22%2C%20%22https%3A//github.com/EnoF/con-rest/pull/45%23discussion_r22851068%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/45%23issuecomment-69584159%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Awesome%20will%20check%20tomorrow%20morning%21%22%2C%20%22created_at%22%3A%20%222015-01-12T15%3A14%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22After%20rebase%20and%20comments%20fixes%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-01-13T07%3A21%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22every%20thing%20ready%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-01-13T09%3A30%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-13T09%3A31%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b28ad95b308b8d7739193c6c6736b5b8d62479ad%20server/mapper.js%2041%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/45%23discussion_r22846246%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22mongoose%20provides%20findByIdAndUpdate%20to%20make%20it%20easier%20for%20you%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-01-13T07%3A16%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-13T09%3A31%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20server/mapper.js%3AL1-95%22%7D%2C%20%22Pull%20b28ad95b308b8d7739193c6c6736b5b8d62479ad%20server/mapper.js%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/EnoF/con-rest/pull/45%23discussion_r22846128%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22pls%20change%20this%20to%20%60mapperScope%60%22%2C%20%22created_at%22%3A%20%222015-01-13T07%3A09%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1508400%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EnoF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-13T09%3A31%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/243319%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dominikkukacka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20server/mapper.js%3AL1-95%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/45#issuecomment-69584159'>General Comment</a></b>
- <a href='https://github.com/EnoF'></a> Awesome will check tomorrow morning!
- <a href='https://github.com/EnoF'></a> After rebase and comments fixes :+1:
- <a href='https://github.com/dominikkukacka'></a> every thing ready :)
- [x] <a href='#crh-comment-Pull b28ad95b308b8d7739193c6c6736b5b8d62479ad server/mapper.js 6'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/45#discussion_r22846128'>File: server/mapper.js:L1-95</a></b>
- <a href='https://github.com/EnoF'></a> pls change this to `mapperScope`
- [x] <a href='#crh-comment-Pull b28ad95b308b8d7739193c6c6736b5b8d62479ad server/mapper.js 41'></a> &nbsp;<b><a href='https://github.com/EnoF/con-rest/pull/45#discussion_r22846246'>File: server/mapper.js:L1-95</a></b>
- <a href='https://github.com/EnoF'></a> mongoose provides findByIdAndUpdate to make it easier for you :)

<a href='https://www.codereviewhub.com/EnoF/con-rest/pull/45?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/EnoF/con-rest/pull/45?approve=1'></a>&nbsp;<a href='https://github.com/EnoF/con-rest/pull/45'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by EnoF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/EnoF/con-rest/pull/45'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by EnoF** [comment]
Awesome will check tomorrow morning!

File: server/mapper.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// con-rest
+// Version: 0.0.1
+//
+// Author: Dominik Kukacka
+// Fork me on Github: https://github.com/EnoF/con-rest
+(function apiScope(mongoose, queue) {
```
Comment (EnoF) [inline server/mapper.js]: pls change this to `mapperScope`

File: server/mapper.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        var id = mongoose.Types.ObjectId(req.params.id);
+        Mapper.findById(id, deferred.makeNodeResolver());
+        deferred.promise.then(function returnCall(call) {
+            res.send(call);
+        });
+        return deferred.promise;
+    }
+
+    function saveMapper(req, res) {
+        var id = mongoose.Types.ObjectId(req.params.id);
```
Comment (EnoF) [inline server/mapper.js]: mongoose provides findByIdAndUpdate to make it easier for you :)

**Comment by EnoF** [comment]
After rebase and comments fixes :+1:

**Comment by dominikkukacka** [comment]
every thing ready :)

File: server/mapper.js (hunk 2 above)
Comment (dominikkukacka) [inline server/mapper.js]: :+1:<a href='#crh-update-none'></a>

**Comment by dominikkukacka** [comment]
:+1:<a href='#crh-update-none'></a>

File: server/mapper.js (hunk 1 above)
Comment (dominikkukacka) [inline server/mapper.js]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**